analyzers/src/SonarAnalyzer.CSharp/Extensions/ISymbolExtensions.cs

```diff
@@ -89,5 +91,28 @@ private static IEnumerable<SyntaxNode> GetDescendantNodes(Location location, Syn
 
             return root.DescendantNodes();
         }
+
+        /// <summary>
+        /// Returns the Identifier token of the first declaring <see cref="SyntaxReference"/>.
+        /// For e.g. an <see cref="IPropertySymbol"/> this is <see cref="PropertyDeclarationSyntax.Identifier"/>.
+        /// </summary>
+        /// <returns>The idenifier token or <see langword="null"/> if there is no DeclaringSyntaxReference or the <see cref="SyntaxNode"/> does not have an Identifier.</returns>
+        public static SyntaxToken? GetFirstIdentifier(this ISymbol symbol) =>
+            symbol.DeclaringSyntaxReferences.FirstOrDefault()?.GetSyntax().NodeIdentifier();
+
+        /// <summary>
+        /// Returns the Identifier tokens of all declaring <see cref="SyntaxReference"/>s.
+        /// For e.g. an <see cref="INamedTypeSymbol"/> this can be a list of partial <see cref="ClassDeclarationSyntax.Identifier"/>.
+        /// </summary>
+        public static ImmutableArray<SyntaxToken> GetIdentifiers(this ISymbol symbol) =>
+            (from r in symbol.DeclaringSyntaxReferences
+             select r.GetSyntax().NodeIdentifier() into i
+             where i != null
+             select (SyntaxToken)i)
+            .ToImmutableArray();
+
+        /// <inheritdoc cref="Helpers.Facade.SyntaxFacade{TSyntaxKind}.NodeIdentifier"/>
+        private static SyntaxToken? NodeIdentifier(this SyntaxNode node)
+            => CSharpFacade.Instance.Syntax.NodeIdentifier(node);
```

Comment:
If you need this extension, then the responsibility should be the other way around. The `Facade` is a dispatcher to language-specific implementation.

Please move the implementation from the `Facade` here. And call this extension from the facade. And do the asymmetric step for VB to keep the codebase consistent.

Reply:
I changed the VB implementation and merged some extensions methods in VB. We should do the same in C# if the implementation is fine in VB.